### PR TITLE
Compiler/0.12

### PR DESCRIPTION
**Pull request by cryogenian**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
-    "test": "pulp test"
+    "test": "pulp test",
+    "ide": "purs ide server"
   },
   "devDependencies": {
-    "pulp": "^11.0.0",
-    "purescript": "^0.11.3",
-    "purescript-psa": "^0.5.0"
+    "pulp": "^12.0.1",
+    "purescript": "slamdata/node-purescript#0.12",
```
Comment (natefaubion) [inline package.json]: 0.12 is on npm now.

File: package.json [hunk 2]
```diff
@@ -2,11 +2,12 @@
   "private": true,
   "scripts": {
     "build": "pulp build -- --censor-lib --strict",
-    "test": "pulp test"
+    "test": "pulp test",
+    "ide": "purs ide server"
```
Comment (natefaubion) [inline package.json:6]: Why was this added?

File: bower.json [hunk 3]
```diff
@@ -23,26 +23,24 @@
     "test"
   ],
   "dependencies": {
-    "purescript-aff": "^4.0.0",
-    "purescript-console": "^3.0.0",
-    "purescript-eff": "^3.1.0",
```
Comment (natefaubion) [inline bower.json:28]: I think we need a `purescript-effect` dependency.

File: bower.json [hunk 4]
```diff
@@ -23,26 +23,24 @@
...
+    "purescript-unsafe-coerce": "^4.0.0",
+    "purescript-variant": "^5.0.0",
+    "purescript-profunctor": "^4.0.0"
   },
   "devDependencies": {
-    "purescript-monad-loops": "^0.4.0",
-    "purescript-control": "^3.3.1",
-    "purescript-minibench": "^1.0.0",
-    "purescript-identity": "^3.1.0"
+    "purescript-monad-loops": "cryogenian/purescript-monad-loops#compiler/0.12",
```
Comment (natefaubion) [inline bower.json]: I think only one combinator is used. Maybe we can replace it's usage with an inline loop.

File: package.json (hunk 2 above)
Comment (cryogenian) [reply package.json:6]: To run local ide server w/o being bound to particular ide bindings implementation. E.g. I have troubles launching ide through emacs and don't have global installation of purescript.

File: bower.json (hunk 3 above)
Comment (cryogenian) [reply bower.json:28]: Okay, I'll add `purescript-effect`. Just wondering why do we need to provide it explicitly? `ps-aff` and `ps-console` have `purescript-effect` as a dependency already.

**Comment by cryogenian** [comment]
+ Removed `monad-loops` dep
+ Changed npm-ps version to 0.12.

File: bower.json (hunk 3 above)
Comment (natefaubion) [reply bower.json:28]: I think console should actually be in dev dependencies. But it's just because I don't like transitive dependencies. If I import it in the library, then it should be in the list of dependencies.

Comment (cryogenian) [reply bower.json:28]: > I think console should actually be in dev dependencies

:+1: Moved

**Comment by natefaubion** [comment]
Thank you so much for helping with this!
